lib/ui/dart_ui.cc

```diff
@@ -34,70 +34,278 @@
 #include "flutter/lib/ui/text/paragraph_builder.h"
 #include "flutter/lib/ui/window/platform_configuration.h"
 #include "third_party/tonic/converter/dart_converter.h"
+#include "third_party/tonic/dart_args.h"
 #include "third_party/tonic/logging/dart_error.h"
 
 using tonic::ToDart;
 
 namespace flutter {
-namespace {
 
-static tonic::DartLibraryNatives* g_natives;
+typedef CanvasImage Image;
+typedef CanvasPathMeasure PathMeasure;
+typedef CanvasGradient Gradient;
+typedef CanvasPath Path;
 
-Dart_NativeFunction GetNativeFunction(Dart_Handle name,
-                                      int argument_count,
-                                      bool* auto_setup_scope) {
-  return g_natives->GetNativeFunction(name, argument_count, auto_setup_scope);
-}
+// List of native static functions used as @FfiNative functions.
+// Items are tuples of ('function_name', 'parameter_count'), where:
+//   'function_name' is the fully qualified name of the native function.
+//   'parameter_count' is the number of parameters the function has.
+//
+// These are used to:
+// - Instantiate FfiDispatcher templates to automatically create FFI Native
+//   bindings.
+//   If the name does not match a native function, the template will fail to
+//   instatiate, resulting in a compile time error.
+// - Resolve the native function pointer associated with an @FfiNative function.
+//   If there is a mismatch between name or parameter count an @FfiNative is
+//   trying to resolve, an exception will be thrown.
+#define FFI_FUNCTION_LIST(V)                                          \
+  /* Constructors */                                                  \
+  V(Canvas::Create, 6)                                                \
```

Comment:
This list should be alphabetized. But is there any way that these lists could go back into the separate files? That seemed easier to manage than this giant list.

Reply:
They are alphabetized within their subsections (e.g. all constructors, all "other").

I don't have a strong preference either way here. This was part of Clement's original refactor.